src/Juvix/Interpreter/InteractionNet/Nets/Default.hs

```diff
@@ -563,9 +596,25 @@ curry2 (nodeF, numNode) (p, numPrim) = do
   linkAll currNode
   deleteRewire [numNode, numPrim] [curr]
 
+curryPrim2 ∷
+  InfoNetwork net (Lang primVal) m ⇒
+  (primVal → primVal → Maybe primVal, Node) →
+  (primVal, Node) →
+  m ()
+curryPrim2 (nodeF, numNode) (p, numPrim) = do
+  incGraphSizeStep (- 1)
+  curr ← newNode (Auxiliary1 $ PrimCurried1 $ nodeF p)
+  let currNode = RELAuxiliary1
+        { node = curr,
+          primary = ReLink numNode Aux2,
+          auxiliary1 = ReLink numNode Aux1
+        }
+  linkAll currNode
+  deleteRewire [numNode, numPrim] [curr]
+
```

Comment:
What is the difference between this and curry2?

Reply:
They should be de-duplicated, we'll get rid of `Primitive`.

Reply:
Noted in https://github.com/cryptiumlabs/juvix/issues/184.